flink-test-utils-parent/flink-connector-testing/src/main/java/org/apache/flink/connectors/test/common/testsuites/SourceTestSuiteBase.java

```diff
@@ -0,0 +1,452 @@
+     *
+     * @param testRecordIterator Iterator of test records
+     * @param resultRecordIterator Iterator of result records consumed by Flink
+     * @param limit Number of records to check
+     */
+    protected void checkSingleSplitRecords(
+            Iterator<T> testRecordIterator, Iterator<T> resultRecordIterator, int limit) {
+        for (int i = 0; i < limit; i++) {
+            T testRecord = testRecordIterator.next();
+            T resultRecord = resultRecordIterator.next();
+            assertEquals(testRecord, resultRecord);
+        }
+        LOG.debug("{} records are validated", limit);
+    }
+
+    /**
+     * Check if records consumed by Flink are identical to test records written into the single
+     * split.
+     *
+     * @param testRecordIterator Iterator of test records
+     * @param resultRecordIterator Iterator of result records consumed by Flink
+     */
+    protected void checkSingleSplitRecords(
+            Iterator<T> testRecordIterator, Iterator<T> resultRecordIterator) {
+        int recordCounter = 0;
+
+        while (testRecordIterator.hasNext()) {
+            T testRecord = testRecordIterator.next();
+            T resultRecord = resultRecordIterator.next();
+            assertEquals(testRecord, resultRecord);
+            recordCounter++;
+        }
+
+        assertFalse(resultRecordIterator.hasNext());
+        LOG.debug("{} records are validated", recordCounter);
+    }
+
+    /**
+     * Check if records consumed by Flink is identical to test records written into multiple splits.
+     *
+     * <p>The order of records across different splits can be arbitrary, but should be identical
+     * within a split.
+     *
+     * @param testData Test data in all splits
+     * @param resultRecordIterator Iterator of result records consumed by Flink
+     */
+    protected void checkMultipleSplitRecords(
+            MultipleSplitTestData testData, Iterator<T> resultRecordIterator) {
+        int recordCounter = 0;
+
+        while (resultRecordIterator.hasNext()) {
+            testData.matchThenNext(resultRecordIterator.next());
+            recordCounter++;
+        }
+
+        assertTrue(
+                testData.hasReachedEnd(),
+                "Records received by Flink is less than records sent to external system.");
+
+        LOG.debug("{} records are validated", recordCounter);
+    }
+
+    /**
+     * Wrapper class for validating test data in multiple splits.
+     *
+     * <p>Each collection has a pointer (iterator) pointing to current checking record. When a
+     * record is received in the stream, it will be compared to all current pointing records in
+     * collections, and the pointer to the identical record will move forward.
+     *
+     * <p>If the stream preserves the correctness and order of records in all splits, all pointers
+     * should reach the end of the collection finally.
+     */
+    public class MultipleSplitTestData {
```

Comment:
static

Reply:
I think this should be a matcher instead. Then we could have
`assertThat(resultIterator, matchesSplits(testRecordCollections));`
If you change it, then it would be nice to have it on top-level for easier reuse.